packages/ua-devtools/src/oapp/config.ts

```diff
@@ -240,84 +240,126 @@ export const configureReceiveLibraryTimeouts: OAppConfigurator = withOAppLogger(
-                to.eid,
-                config.sendConfig.ulnConfig
-            )
+            if (config.sendConfig.ulnConfig != null) {
+                // We ask the endpoint SDK whether this config has already been applied
+                //
+                // We need to ask not for the final config formed of the default config and the app config,
+                // we only need to check the app config
+                const hasUlnConfig = await endpointSdk.hasAppUlnConfig(
+                    from.address,
+                    currentSendLibrary,
+                    to.eid,
+                    config.sendConfig.ulnConfig
+                )
 
-            if (!hasUlnConfig) {
-                const newSetConfigs: SetConfigParam[] = await endpointSdk.getUlnConfigParams(currentSendLibrary, [
-                    { eid: to.eid, ulnConfig: config.sendConfig.ulnConfig },
-                ])
-
-                // Updates map with new configs for that OApp and Send Library
-                const setConfigsByLibrary = setConfigsByEndpointAndLibrary.getOrElse(from, () => new Map())
-                const existingSetConfigs = setConfigsByLibrary.get(currentSendLibrary) ?? []
-                setConfigsByEndpointAndLibrary.set(
-                    from,
-                    setConfigsByLibrary.set(currentSendLibrary, [...existingSetConfigs, ...newSetConfigs])
+                logger.verbose(
+                    `Checked Uln configuration for ${formatOmniVector({ from, to })}: ${printBoolean(hasUlnConfig)}`
                 )
+
+                if (!hasUlnConfig) {
+                    const newSetConfigs: SetConfigParam[] = await endpointSdk.getUlnConfigParams(currentSendLibrary, [
+                        { eid: to.eid, ulnConfig: config.sendConfig.ulnConfig },
+                    ])
+
+                    // Updates map with new configs for that OApp and Send Library
+                    const setConfigsByLibrary = setConfigsByEndpointAndLibrary.getOrElse(from, () => new Map())
+                    const existingSetConfigs = setConfigsByLibrary.get(currentSendLibrary) ?? []
+                    setConfigsByEndpointAndLibrary.set(
+                        from,
+                        setConfigsByLibrary.set(currentSendLibrary, [...existingSetConfigs, ...newSetConfigs])
+                    )
+
+                    const updatedConfigList =
+                        setConfigsByEndpointAndLibrary.getOrElse(from, () => new Map()).get(currentSendLibrary) ?? []
+                    const updatedConfigListCsv = updatedConfigList
+                        .map(
+                            (setConfigParam) =>
+                                '{configType: ' + setConfigParam.configType + ', config: setConfigParam.config}'
+                        )
+                        .join(', ')
+                    logger.verbose(
+                        `Set Uln configuration ${updatedConfigListCsv} for ${formatOmniVector({ from, to })}`
```

Comment:
This is the change , along with the other log above (the rest of the function was just indented)